codebuild/bin/s2n_codebuild.sh

```diff
@@ -47,7 +47,15 @@ if [[ "$OS_NAME" == "linux" && "$TESTS" == "valgrind" ]]; then
     # We have to output something every 9 minutes, as some test may run longer than 10 minutes
     # and will not produce any output
     while sleep 9m; do echo "=====[ $SECONDS seconds still running ]====="; done &
-    S2N_DEBUG=true make -j $JOBS valgrind
+
+    if [[ "$GCC_VERSION" == "9" && "$LIBCRYPTO_ROOT" == "openssl-1.1.1" ]]; then
```

Comment:
We should enable the pedantic testing for all compilers. We will need to be careful to update this if we decide to get rid of GCC9

Reply:
Can we make this error if gcc version 9 isn't available so we don't forget to update this?

Reply:
Not sure how we would add that check since these values get populated from codebuild. We might need to be careful. Maybe a comment somewhere.

I also wonder if this is something we care about. We dont typicially remove GCC compilers to remain backwards compatible and would probably just add another.

Reply:
@dougch thoughts ^